app/models/embed/purl.rb

```diff
@@ -208,6 +224,11 @@ def files
         end
       end
 
+      def size
+        # need to guard against file.size being nil when summing
+        files.filter_map { |file| file.size unless file.size.nil? }.sum
```

Comment:
I *think* the `unless file.size.nil?` is not needed here; `#filter_map` will automatically drop all `nil` values.

Reply:
yep, works! thanks 🙇